src/Spark.Mongo/Store/MongoFhirStore.cs

```diff
@@ -92,7 +92,42 @@ public async Task<Entry> GetAsync(IKey key)
                 ?.ToEntry();
         }
 
-        public IList<Entry> Get(IEnumerable<IKey> identifiers)
+        private IFindFluent<BsonDocument, BsonDocument> AddProjection(IFindFluent<BsonDocument, BsonDocument> queryable, IEnumerable<string> elements)
+        {
+            if (elements != null && elements.Any())
+            {
+                // add metadata
+                var projection = Builders<BsonDocument>.Projection
+                .Include(Field.PRIMARYKEY)
+                .Include(Field.REFERENCE)
+                .Include(Field.WHEN)
+                .Include(Field.STATE)
+                .Include(Field.VERSIONID)
+                .Include(Field.TYPENAME)
+                .Include(Field.METHOD)
+                .Include(Field.TRANSACTION)
+
+                // add required fields
+                // TODO: Maybe we should get required fields from the StructureDefinition of Resource. 
```

Comment:
Nice,But need `IsModifier `?

Reply:
The way I'm interpreting the documentation, yes: http://hl7.org/fhir/search.html#elements
> Clients SHOULD list all mandatory and modifier elements in a resource as part of the list of elements.



Also I noticed this from the same section in the spec:
> Servers SHOULD mark the resources with the tag [SUBSETTED](http://hl7.org/fhir/R4/v3/ObservationValue/cs.html#v3-ObservationValue-SUBSETTED) to ensure that the incomplete resource is not actually used to overwrite a complete resource.

This can be added like this:
```
resource.Meta.Tag.Add(new Coding
{
    System = "http://terminology.hl7.org/CodeSystem/v3-ObservationValue",
    Code = "SUBSETTED",
    Display = "subsetted",
});
```

Reply:
OK,I get it.
I will try to modify it.